src/RoboSaverVirtualModule.sol

```diff
@@ -335,6 +373,12 @@ contract RoboSaverVirtualModule is
     /// @param _deficit The amount of $EURe to withdraw from the pool
     /// @return request_ The exit pool request as per Balancer's interface
     function _poolWithdrawal(uint256 _deficit) internal returns (IVault.ExitPoolRequest memory request_) {
+        /// @dev Payload 1: Unstake and claim all pending rewards from the Aura gauge
+        uint256 maxBPTAmountIn = _deficit * MAX_BPS * 1e18 / (MAX_BPS - slippage) / BPT_STEUR_EURE.getRate();
```

Comment:
this is maybe a bit of psycho comment, but we are calculating onchain the maxbptin, but we are not checking at any moment after the offchain calculation here that indeed the `CARD` has sufficient bpt balance to cover the `maxBPTAmountIn` maybe can be a health check just below this L377

otherwise may revert and could be trickier to debug reasoning

Reply:
also i noticed that the calculation for the `deficit` on L298 consider the balances of the native balancer gauge + the reward aura contract balance, while in here the withdrawal only happens from the reward aura contract itself, so i think both calculation should match

Reply:
added in af97d66ed127dd1c23359cef82c39fc68c246f06